CMakeLists.txt

```diff
@@ -51,12 +51,20 @@ FIND_PACKAGE(PkgConfig)
 FIND_PACKAGE(Threads REQUIRED)
 
 IF ((CMAKE_SYSTEM_NAME STREQUAL "Darwin") AND NOT (DEFINED OPENSSL_ROOT_DIR OR DEFINED ENV{OPENSSL_ROOT_DIR}))
-    MESSAGE(STATUS "*************************************************************************\n"
-                "   * Setting OPENSSL_ROOT_DIR to /usr/local/opt/openssl. On macOS, OpenSSL *\n"
-                "   * should be installed using homebrew or OPENSSL_ROOT_DIR must be set to *\n"
-                "   * the path that has OpenSSL installed.                                  *\n"
-                "   *************************************************************************")
-    SET(OPENSSL_ROOT_DIR "/usr/local/opt/openssl")
+    EXECUTE_PROCESS(
+        COMMAND brew --prefix openssl
+        OUTPUT_VARIABLE OPENSSL_ROOT_DIR
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+        ERROR_QUIET
+    )
+    IF (NOT EXISTS ${OPENSSL_ROOT_DIR})
+        MESSAGE(STATUS "*************************************************************************\n"
+                    "   * Setting OPENSSL_ROOT_DIR to /usr/local/opt/openssl. On macOS, OpenSSL *\n"
+                    "   * should be installed using homebrew or OPENSSL_ROOT_DIR must be set to *\n"
+                    "   * the path that has OpenSSL installed.                                  *\n"
+                    "   *************************************************************************")
+        SET(OPENSSL_ROOT_DIR, "/usr/local/opt/openssl")
```

Comment:
Thank you for the changes. Would you mind changing the fallback path from "/usr/local/opt/openssl" to "/usr/local"? I'd assume that `/usr/local` would make more sense here.

Other than that, the PR looks good to me.

Reply:
Agreed, it looks better. Done ea8cf76b4c83aad95e7599db46a580660efe8383 

(also removed a comma, which probably I added as a mistake 🙇 )